L03-generics/src/main/java/ru/otus/homework/CustomerService.java

```diff
@@ -0,0 +1,36 @@
+package ru.otus.homework;
+
+import java.util.Map;
+import java.util.SortedMap;
+import java.util.TreeMap;
+
+public class CustomerService {
+
+    private final SortedMap<Customer, String> customerData = new TreeMap<>();
```

Comment:
```suggestion
    private final NavigableMap<Customer, String> customerData = new TreeMap<>();
```

Здесь достаточно `NavigableMap`

Reply:
поправил